src/parser.c

```diff
@@ -130,11 +130,19 @@ void set_error(parserstate *state, token tok, bool syntax_error, const char *fmt
     return;
   }
 
-  char *message;
-
   va_list args;
+
+  // First get the required length
+  va_start(args, fmt);
+  int len = vsnprintf(NULL, 0, fmt, args);
+  va_end(args);
```

Comment:
Self explanatory, a better variable name would remove the need for the comment

```suggestion
  va_start(args, fmt);
  int length = vsnprintf(NULL, 0, fmt, args);
  va_end(args);
```